cloudflare/resource_cloudflare_custom_ssl.go

```diff
@@ -0,0 +1,284 @@
+}
+
+func resourceCloudflareCustomSslRead(d *schema.ResourceData, meta interface{}) error {
+
+	client := meta.(*cloudflare.API)
+	zoneID := d.Get("zone_id").(string)
+	certID := d.Id()
+
+	record, err := client.SSLDetails(zoneID, certID)
+	if err != nil {
+		log.Printf("[WARN] Removing record from state because it's not found in API")
+		d.SetId("")
+		return nil
+	}
+
+	d.SetId(record.ID)
+	d.Set("hosts", record.Hosts)
+	d.Set("issuer", record.Issuer)
+	d.Set("signature", record.Signature)
+	d.Set("status", record.Status)
+	d.Set("bundle_method", record.BundleMethod)
+	d.Set("uploaded_on", record.UploadedOn.Format(time.RFC3339Nano))
+	d.Set("expires_on", record.ExpiresOn.Format(time.RFC3339Nano))
+	d.Set("modified_on", record.ModifiedOn.Format(time.RFC3339Nano))
+	d.Set("priority", record.Priority)
+	return nil
+}
+
+func resourceCloudflareCustomSslDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	zoneID := d.Get("zone_id").(string)
+	certID := d.Id()
+
+	log.Printf("[DEBUG] Deleting SSL cert %s for zone %s", certID, zoneID)
+
+	err := client.DeleteSSL(zoneID, certID)
+	if err != nil {
+		errors.Wrap(err, "failed to delete custom ssl cert setting")
+	}
+	return nil
+}
+
+func resourceCloudflareCustomSslImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	// split the id so we can lookup
+	idAttr := strings.SplitN(d.Id(), "/", 2)
+
+	if len(idAttr) != 2 {
+		return nil, fmt.Errorf("invalid id (\"%s\") specified, should be in format \"zoneID/certID\"", d.Id())
+	}
+
+	zoneID, certID := idAttr[0], idAttr[1]
+
+	log.Printf("[DEBUG] Importing Cloudflare Custom SSL Cert: id %s for zone %s", certID, zoneID)
+
+	d.Set("zone_id", zoneID)
+	d.SetId(certID)
+
+	resourceCloudflareCustomSslRead(d, meta)
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func expandToZoneCustomSSLPriority(d *schema.ResourceData) ([]cloudflare.ZoneCustomSSLPriority, error) {
+	data, dataOk := d.GetOk("custom_ssl_priority")
+	log.Printf("[DEBUG] Custom SSL priority found in config: %#v", data)
+	//type mt map[string]interface{}
+	var mtSlice []cloudflare.ZoneCustomSSLPriority
+	if dataOk {
+		for _, innerData := range data.([]interface{}) {
+			newData := make(map[string]interface{})
+			for id, value := range innerData.(map[string]interface{}) {
+				switch idName := id; idName {
+				case "id":
+					newValue := value.(string)
+					//newData[id] = newValue
+					newData["ID"] = newValue
+				case "priority":
+					newValue := value.(int)
+					newData[id] = newValue
+				default:
+					newValue := value
+					newData[id] = newValue
+				}
+			}
+			zcsp := cloudflare.ZoneCustomSSLPriority{}
+			zcspJson, err := json.Marshal(newData)
+			if err != nil {
+				return mtSlice, fmt.Errorf("Failed to create custom ssl priorities: %s", err)
+			}
+			// map -> json -> struct
+			json.Unmarshal(zcspJson, &zcsp)
+			//return mtSlice, fmt.Errorf("FAILURE: %#v %#v", newData, zcsp)
```

Comment:
If this isn't needed, let's get rid of it.

```suggestion
```

Reply:
👍